src/caselawclient/xquery/set_metadata_date.xqy

```diff
@@ -4,13 +4,20 @@ declare namespace akn = "http://docs.oasis-open.org/legaldocml/ns/akn/3.0";
 declare variable $uri as xs:string external;
 declare variable $content as xs:string external;
 
+(: If at least one FRBRWork/FRBRdate element exists then...:)
 if (fn:boolean(
-cts:search(doc($uri),
+cts:search(doc($uri)/akn:akomaNtoso/akn:judgment/akn:meta/akn:identification/akn:FRBRWork,
 cts:element-query(xs:QName('akn:FRBRdate'),cts:and-query(()))))) then
-    xdmp:node-replace(
-    document($uri)/akn:akomaNtoso/akn:judgment/akn:meta/akn:identification/akn:FRBRWork/akn:FRBRdate,
-    <akn:FRBRdate date="{$content}" name="judgment" xmlns="http://docs.oasis-open.org/legaldocml/ns/akn/3.0"/>)
+    (: ...iterate over the FRBRdate element(s) :)
+    for $frbr_date in doc($uri)/akn:akomaNtoso/akn:judgment/akn:meta/akn:identification/akn:FRBRWork/akn:FRBRdate
+        (: Store the existing @name attribute value :)
+        let $attribute_name := $frbr_date/@name
+        (: Replace the exisitng FRBRdate element with a new one, using the same @name attribute :)
+        return xdmp:node-replace(
+        document($uri)/akn:akomaNtoso/akn:judgment/akn:meta/akn:identification/akn:FRBRWork/akn:FRBRdate[@name = $attribute_name],
```

Comment:
can this be `$frbr_date` ?
Is there a difference between doc($uri) and document($uri) ?